weld/src/main/java/org/jboss/as/weld/deployment/processors/ProbeDeploymentProcessor.java

```diff
@@ -117,10 +136,23 @@ public void deploy(DeploymentPhaseContext phaseContext) throws DeploymentUnitPro
         }
         webMetaData.getServletMappings().add(PROBE_SERVLET_MAPPING);
 
-        // Injection into servlet
+        // Filter
+        if (webMetaData.getFilters() == null) {
+            webMetaData.setFilters(new FiltersMetaData());
+        }
+        webMetaData.getFilters().add(PROBE_FILTER);
+
+        // Filter mapping
+        if (webMetaData.getFilterMappings() == null) {
+            webMetaData.setFilterMappings(new ArrayList<FilterMappingMetaData>());
+        }
+        webMetaData.getFilterMappings().add(PROBE_FILTER_MAPPING);
```

Comment:
If we add the filter as the last one in the chain then any other filter that is intercepted (each of them by default), will cause InvocationMonitor to be started thus we lose HTTP request information. We should therefore register it as the first one.